protocol/communities_messenger_test.go

```diff
@@ -3585,3 +3654,27 @@ func (s *MessengerCommunitiesSuite) TestHandleImport() {
 	s.Require().NotNil(chat)
 	s.Require().Equal(0, int(chat.UnviewedMessagesCount))
 }
+
+func (s *MessengerCommunitiesSuite) TestJoinedCommunityMembersSharedAddress() {
+	community := s.createCommunity()
+	s.advertiseCommunityTo(community, s.alice)
+	s.advertiseCommunityTo(community, s.bob)
+
+	s.joinCommunity(community, s.alice, AlicePassword)
+	s.joinCommunity(community, s.bob, BobPassword)
+
+	joinedCommunities, err := s.admin.communitiesManager.Joined()
+	s.Require().NoError(err)
+	s.Require().Len(joinedCommunities, 1)
+	community = joinedCommunities[0]
+	s.Require().NotNil(community)
+
+	s.Require().Equal(3, community.MembersCount())
+
+	for pubKey, member := range community.Members() {
+		if pubKey != common.PubkeyToHex(&s.admin.identity.PublicKey) {
+			s.Require().Len(member.RevealedAccounts, 1)
+			s.Require().Equal(member.RevealedAccounts[0].Address, walletAddress)
+		}
```

Comment:
Since we're iterating over members here, do we expect both, `bob` and `alice` to reveal the exact same `walletAddress`?

Reply:
Yes, it is done consciously. The problem is in `communities/manager.go:1860`:
```
	recovered, err := m.accountsManager.Recover(recoverParams)
	if err != nil {
		return nil, err
	}
	if recovered.Hex() != revealedAccount.Address {
```
We can have different wallet addresses in the test, but that requires forging that address every time we ask to join with a valid wallet address, and that's a pretty painful change. Moreover, tests becoming *more* fragile, because we are trying to fake behaviour of an object, that is far away from issue's scope

Reply:
The right way i see is to change `AccountManager::Recover` api to embed check inside